x/pocketcore/keeper/service_test.go

```diff
@@ -66,25 +72,157 @@ func TestKeeper_HandleRelay(t *testing.T) {
-	defer gock.Off() // Flush pending mocks after test execution
-
-	gock.New("https://www.google.com:443").
-		Post("/").
-		Reply(200).
-		BodyString("bar")
 
 	mockCtx.On("KVStore", keeper.storeKey).Return(ctx.KVStore(keeper.storeKey))
 	mockCtx.On("KVStore", keys["pos"]).Return(ctx.KVStore(keys["pos"]))
 	mockCtx.On("KVStore", keys["params"]).Return(ctx.KVStore(keys["params"]))
 	mockCtx.On("KVStore", keys["application"]).Return(ctx.KVStore(keys["application"]))
-	mockCtx.On("BlockHeight").Return(ctx.BlockHeight())
-	mockCtx.On("PrevCtx", int64(976)).Return(ctx, nil)
+	mockCtx.On("BlockHeight").Return(currentBlockHeight)
+	mockCtx.On("PrevCtx", relayProofSessionBlockHeight).Return(ctx, nil)
 	mockCtx.On("PrevCtx", keeper.GetLatestSessionBlockHeight(mockCtx)).Return(ctx, nil)
 	mockCtx.On("Logger").Return(ctx.Logger())
+	return
+}
+
+func TestKeeper_HandleRelay(t *testing.T) {
+	defer gock.Off() // Flush pending mocks after test execution
 
+	// Each session block is 25 blocks by default genesis params found in DefaultSessionBlocktime, pocket-core/x/nodes/types/params.go
+
+	// tc - height is same as session block
+	tc := relayTc{
+		blockHeight:                  976,
+		relayProofSessionBlockHeight: 976,
+	}
+	gock.New("https://www.google.com:443").
+		Post("/").
+		Reply(200).
+		BodyString("bar")
+	keeper, mockCtx, validRelay := setupHandleRelayTest(t, tc.blockHeight, tc.relayProofSessionBlockHeight)
 	resp, err := keeper.HandleRelay(mockCtx, validRelay)
 	assert.Nil(t, err, err)
 	assert.NotNil(t, resp)
 	assert.NotEmpty(t, resp)
 	assert.Equal(t, resp.Response, "bar")
 }
+
+func TestKeeper_HandleRelayWithinTolerance(t *testing.T) {
+	defer gock.Off() // Flush pending mocks after test execution
+
+	// Each session block is 25 blocks by default genesis params found in DefaultSessionBlocktime, x/nodes/types/params.go
+	// SessionBlockSyncAllowance is 1 session block by default found in DefaultSessionBlockSyncAllowance, types/config.go
+
+	tcs := []relayTc{
+		// tc 1 - relay proof session height matches the same session block (max height boundary)
+		{
+			blockHeight:                  976,
+			relayProofSessionBlockHeight: 976,
+		},
+		// tc2 - relay proof session height is within tolerance (min height boundary)
+		{
+			blockHeight:                  976,
+			relayProofSessionBlockHeight: 951,
+		},
+	}
+
+	for _, tc := range tcs {
+		gock.New("https://www.google.com:443").
+			Post("/").
+			Reply(200).
+			BodyString("bar")
+		keeper, mockCtx, validRelay := setupHandleRelayTest(t, tc.blockHeight, tc.relayProofSessionBlockHeight)
+		resp, err := keeper.HandleRelay(mockCtx, validRelay)
+		assert.Nil(t, err, err)
+		assert.NotNil(t, resp)
+		assert.NotEmpty(t, resp)
+		assert.Equal(t, resp.Response, "bar")
+	}
+}
+
+func TestKeeper_HandleRelayOutsideTolerance(t *testing.T) {
+	// Each session block is 25 blocks by default genesis params found in DefaultSessionBlocktime, pocket-core/x/nodes/types/params.go
```

Comment:
Is this true given that we're modifying `ClientBlockSyncAllowance` below? Ditto for other tests.

Reply:
Yes, session block intervals is an on chain parameter, whereas `ClientBlockSyncAllowance` is a client specific parameter.

Reply:
I agree `ClientBlockSyncAllowance` is confusing.  As @PoktBlade says, it's client-specific and stored in config.json, which means a node runner can set it to a big number like 100 or 100000000 to make `RelayMeta.Validate` always pass.

But, it doesn't impact the Pocket network at all because as @PoktBlade mentions below, relays are handled just locally.  From app or the portal's perspective, the relay result depends on which node the relay is dispatched to, but it doesn't impact the reward mechanism until the node submits a claim tx.  And, once a claim tx is submitted, what the network cares about is whether a relay is processed by a valid node in the corresponding session or not.  It doesn't care whether the node was synced or not when the relay was processed.

Anyway, it makes sense to me that we eliminate the impact of `ClientBlockSyncAllowance` by setting it to a big number like 100.